api/mod/tests/unit/test_module.py

```diff
@@ -3,6 +3,76 @@
 """
 
 
-def test_unit() -> None:
-    """A sample unit test"""
-    assert True
+import json
+from pathlib import Path
+
+import pytest
+from pyfakefs.fake_filesystem_unittest import Patcher
+from pytest_mock import MockerFixture
+
+import mod.src.app
+from mod.src.app import Annotation
+
+
+@pytest.fixture
+def annotation() -> Annotation:
+    return Annotation(
+        src="test.jpg",
+        label="Label",
+        hash="somehash",
+        competency="Prof. Dr. Med.",
+        is_attentive=True,
+    )
+
+
+class TestAnnotation:
+    def test_absolute_src(self, annotation: Annotation, mocker: MockerFixture) -> None:
+        mocker.patch("mod.src.settings.SETTINGS.data_folder", Path("mock_folder"))
+        assert Path("mock_folder/test.jpg") == annotation.absolute_src
+
+    def test_file_exists(self, annotation: Annotation) -> None:
+        with Patcher(modules_to_reload=[mod.src.settings, mod.src.app]) as patcher:
+            assert not annotation.file_exists()
+            patcher.fs.create_file(
+                mod.src.settings.SETTINGS.data_folder.joinpath("test.jpg")
+            )
+            assert annotation.file_exists()
```

Comment:
Yes we tried to mock the whole filesystem and replace it with a fake filesystem that lives in-memory. There is not really a need for it but it sounded good 😅
We could otherwise override the data_folder setting with a temporary folder.
Actually I think this would make the tests easier because pyfakefs had problems with mocking pathlib (that is why the patchers are needed) and the changes resulting in that maybe outweigh the benefits of pyfakefs...

Reply:
Understand, would have been cool. But concerning the overhead (compared to the ~2 lines required currently in the integration tests) I don't really feel it's all worth it.